### PR TITLE
case importer job poll refactor

### PR DESCRIPTION
**Pull request by snopoke**
@biyeun

### CODE REVIEW

**Comment by czue** [comment]
build fail looks like it might be related
```
======================================================================
ERROR: testImportNone (corehq.apps.importer.tests.ImporterTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/importer/tests.py", line 94, in testImportNone
    self.assertEqual('EXPIRED', res['error'])
KeyError: 'error'
```

**Comment by czue** [comment]
the case importer is scary code. :ghost:

**Comment by snopoke** [comment]
:white_check_mark:

File: corehq/ex-submodules/soil/exceptions.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 
 
 class TaskFailedError(Exception):
-    pass
+    def __init__(self, errors, *args, **kwargs):
```
Comment (czue) [inline corehq/ex-submodules/soil/exceptions.py]: will making `errors` required not cause [empty args creation](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/soil/util.py#L40) to fail?

Comment (snopoke) [inline corehq/ex-submodules/soil/exceptions.py]: Indeed it will, well spotted. Thanks.

**Comment by snopoke** [comment]
@czue updated

**Comment by czue** [comment]
(build fail not real)

**Pull request merged**